src/jit/morph.cpp

```diff
@@ -14975,22 +14974,34 @@ bool Compiler::fgFoldConditional(BasicBlock* block)
 
                 if (bUpdated != nullptr)
                 {
+                    BasicBlock::weight_t newMinWeight;
+                    BasicBlock::weight_t newMaxWeight;
+
                     flowList* edge;
                     // Now fix the weights of the edges out of 'bUpdated'
                     switch (bUpdated->bbJumpKind)
                     {
                         case BBJ_NONE:
-                            edge                  = fgGetPredForBlock(bUpdated->bbNext, bUpdated);
-                            edge->flEdgeWeightMax = bUpdated->bbWeight;
+                            edge         = fgGetPredForBlock(bUpdated->bbNext, bUpdated);
+                            newMaxWeight = bUpdated->bbWeight;
+                            newMinWeight = min(edge->edgeWeightMin(), newMaxWeight);
+                            edge->setEdgeWeights(newMinWeight, newMaxWeight);
```

Comment:
These assignments of flEdgeWeightMax  could violate the requirements that  min-weight <= maxWeight